tests/test_std.rs

```diff
@@ -558,6 +558,51 @@ fn concat_non_empty() {
     assert_eq!(data.into_iter().concat(), vec![1,2,3,4,5,6,7,8,9])
 }
 
+
+#[test]
+fn test_combinations_by_len() {
+    let it = it::combinations_by_len(1..3, 0);
+    it::assert_equal(it, vec![
+        vec![],
+    ]);
+
+    let it = it::combinations_by_len(1..3, 1);
+    it::assert_equal(it, vec![
+        vec![1],
+        vec![2],
+    ]);
+
+    let it = it::combinations_by_len(1..3, 2);
+    it::assert_equal(it, vec![
+        vec![1, 2],
+    ]);
+
+    assert!(it::combinations_by_len(1..3, 3).next().is_none());
+
+    let it = it::combinations_by_len(1..5, 2);
+    it::assert_equal(it, vec![
+        vec![1, 2],
+        vec![1, 3],
+        vec![2, 3],
+        vec![1, 4],
+        vec![2, 4],
+        vec![3, 4],
+        ]);
+
+    let mut it = it::combinations_by_len(1..,3);
+    assert_eq!(it.next(), Some(vec![1,2,3]));
+    assert_eq!(it.next(), Some(vec![1,2,4]));
+    assert_eq!(it.next(), Some(vec![1,3,4]));
+    assert_eq!(it.next(), Some(vec![2,3,4]));
+    assert_eq!(it.next(), Some(vec![1,2,5]));
+    assert_eq!(it.next(), Some(vec![1,3,5]));
+    assert_eq!(it.next(), Some(vec![1,4,5]));
+    assert_eq!(it.next(), Some(vec![2,3,5]));
+    assert_eq!(it.next(), Some(vec![2,4,5]));
+    assert_eq!(it.next(), Some(vec![3,4,5]));
+    assert_eq!(it.next(), Some(vec![1,2,6]));
+}
+
```

Comment:
Should we add some tests that actually work on infinite iterators?

Reply:
The last test does so (iterator is `(1..)` by 3).
Sadly in test we can test only for a finite number of elements! I have decided to stop on 11.

Reply:
Ah, thank you, I misread that.